### DISCUSSION TITLE
Does Decap support authentication with GitHub as the identity provider? (answered)

### DISCUSSION THREAD

**Discussion by alexanderankin** [decaporg/decap-cms#6729] [Q&A]
I am wondering if there is a simple way to use github's api to author commits from the frontend with just a github username and password (or personal auth token with the right repo permissions)?

Just researching what the capabilities are now. I saw the docs allow for something [integrated with netlify here](https://decapcms.org/docs/add-to-your-site/#add-the-netlify-identity-widget) but I was wondering if it had been done with GitHub as well (or how much effort I could expect it to take for me).

edit: also noticed the ["GitHub Backend" page](https://decapcms.org/docs/github-backend/)

## answer: https://github.com/decaporg/decap-cms/issues/663#issuecomment-335023723

**Comment by airtonix** [comment]
> I am wondering if there is a simple way to use github's api to author commits from the frontend with just a github username and password (or personal auth token with the right repo permissions)?

yes but due to how oauth works, you should not expose the backchannel secrets used to verify a user. Thus you need to provide two endpoints and a configuration change in order for your custom setup to work.

With NextJS and vercel/cloudflare-pages/etc this is as trivial as adding the the below setup and configuring some server side secrets

👉🏻  https://gist.github.com/airtonix/293705c3d67ed6735f0e5e540d73b67e

> edit: also noticed the ["GitHub Backend" page](https://decapcms.org/docs/github-backend/)

It'd be great if the the above gist was officially a monorepo package from `@decap/github-oauth-routes` (or something)
